pallets/sto/src/lib.rs

```diff
@@ -354,7 +354,7 @@ decl_module! {
             with_transaction(|| {
                 <Portfolio<T>>::unlock_tokens(&fundraiser.offering_portfolio, &fundraiser.offering_asset, &investment_amount)?;
 
```

Comment:
We need more changes in the STO module. The module assumes that if the `base_affirm_instruction` is successful, the instruction has settled and it updates the purchase stats a few lines below. 

Now, the formation can be successful but the settlement can fail in the next block. We need to handle the purchase stats accordingly. There are a few edge cases to think about like what if multiple people try to invest such that only one can go through.